internal/controllers/topology/cluster/desired_state.go

```diff
@@ -724,14 +737,17 @@ func templateToObject(in templateToInput) (*unstructured.Unstructured, error) {
 	labels[clusterv1.ClusterLabelName] = in.cluster.Name
 	labels[clusterv1.ClusterTopologyOwnedLabel] = ""
 
+	templateLabels := mergeMap(labels, in.labels)
```

Comment:
Given that we are accepting labels and annotations in input, I think we should drop adding labels in this func so there will be a unique point where we define all the labels for the new objects